modules/database/src/ioc/db/dbChannel.c

```diff
@@ -462,15 +493,36 @@ dbChannel * dbChannelCreate(const char *name)
     if (status)
         goto finish;
 
-    chan = freeListCalloc(dbChannelFreeList);
-    if (!chan)
-        goto finish;
     cname = malloc(strlen(name) + 1);
     if (!cname)
         goto finish;
 
-    strcpy(cname, name);
+    if (strcmp(dbEntry.pflddes->name, "TIME")) {
```

Comment:
Did you explore re-writing the PV name string as it is copied?  eg. to match on strings ending with ".TIME" and append some `{"ts":...` before further parsing?  It seems like this would avoid much special case handling of "ts".

Reply:
Yes. I chose the current implementation for two reasons:
- I see no reason not to allow subscribing to `test:counter.TIME{"dec": {"n": 3}}`. Simply appending `{"ts": ...}` would break that.
- I like calling `pvNameLookup()` first because it ensures that `name` has the form I assume. I do not know what assumptions I can make about `name` by the time it reaches `dbChannelCreate()`.

Reply:
I am thinking the same thing wrt. other filters.  My main motivation though is code simplification.  To minimize the special case handling of the `ts` filter.

Since the channel name string is always copied, it seems reasonable to do so a little earlier (before calling `pvNameLookup()`), where some re-writing would be less invasive.  Something like the following:

```c
...
/* implied default filter */
#define TIMEfilt ".TIME{\"ts\":..."
size_t namelen = strlen(name);
if(name>=5 && strcmp(name+namelen-5u, ".TIME")==0) { /* aka. if name ends with ".TIME" */
    cname = malloc(namelen-5u+sizeof(TIMEfilt));
    memcpy(cname, name, namelen-5u);
    memcpy(cname+namelen-5u, TIMEfilt, sizeof(TIMEfilt));
} else {
    cname = malloc(namelen+namelen+1u);
    stcpy(cname, name);
}
pname = cname;

status = pvNameLookup(&dbEntry, &pname); /* ... handle as before ... */
...
```

Reply:
Sorry for taking this long to respond, I've been quite busy.

I followed your approach to see what the end result looks like, and I'm not entirely happy. Personally, I hate working with strings in C because it takes a *lot* of care to get even trivial things right, and reading such code is not my cup of tea. That's why I still prefer the previous approach where `pvNameLookup()` was reused to parse the string, the name was merely shortened and the string code was straightforwad. 

That's just my opinion FYI, of course; being the maintainer, it's up to you to decide which approach you find more maintainable. The change is in commit 7ba4b22

Reply:
In your latest change, it looks like you are trying to be more general than I was thinking by "merging" two JSON filter expressions. If I'm misunderstanding this code, please give an example of what a rewritten PV name should look like before and after.

> ... I see no reason not to allow subscribing to `test:counter.TIME{"dec": {"n": 3}}` ...

I am strongly inclined towards strictness in parsing.  It is much easier to later relax restrictions than it is to later impose them.

With this in mind, I don't see this generality as necessary at present.  It seems reasonable to me to assume that someone explicitly giving a filter for .TIME knows what this entails.  Further, an explicit filter spec may well include "TS" filter, in which case automatically prepending a second instance may be surprising or unhelpful.

> ... Personally, I hate working with strings in C ...

I absolutely agree, this is why for me short and simple are important criteria.